worker.py: Avoid letting a job get stuck if exception occurs

JonathonHall-Purism opened the pull request:
Some instances of stuck jobs were observed recently for PureOS.  From the logs, I think a Python exception may have occurred after the build completed but before the artifacts were uploaded.  I can't tell what might have caused that exception, if it did occur.

This change would ensure that a 'rejected' status is sent if this occurs, rather than leaving the job stuck with no result.

I applied this change to fennel.pureos.net (the new worker) to try to identify what was causing the stuck jobs, but this never happened again.  No jobs got stuck after that and this exception code was never hit, so I can't identify the root cause.

JonathonHall-Purism committed:
worker.py: Avoid letting a job get stuck if exception occurs

Some instances of stuck jobs were observed recently for PureOS.  From
the logs, I think a Python exception may have occurred after the build
completed but before the artifacts were uploaded.  I can't tell what
might have caused that exception, if it did occur.

This change would ensure that a 'rejected' status is sent if this
occurs, rather than leaving the job stuck with no result.

I applied this change to fennel.pureos.net (the new worker) to try to
identify what was causing the stuck jobs, but this never happened
again.  No jobs got stuck after that and this exception code was never
hit, so I can't identify the root cause.

Signed-off-by: Jonathon Hall <jonathon.hall@puri.sm>

JonathonHall-Purism commented:
I have not properly tested this change - I don't have a test environment set up, and this never happened again after I applied the change to fennel.pureos.net to see what was causing these jobs to get stuck this way.

The jobs were very old (there had been no amd64 worker for months), so it is possible that this would not happen if the job queue is being processed regularly.

So with that in mind, although it seems like a good robustness improvement to avoid leaving a job stuck if this happens, I understand if you'd rather reject this and see if it ever happens again or can be reproduced.

ximion commented on spark/worker.py:
```diff
@@ -109,7 +109,7 @@ def _run_job(self, job):
 
                     tb = traceback.format_exc()
-                    jlog.write(tb)
+                    log.write(tb)
```
This looks unintentional...

JonathonHall-Purism replied:
Yeah, whoops, I will revert that

JonathonHall-Purism committed:
spark/worker.py: Revert errant jlog/log change, fix traceback log

jlog was accidentally changed to log in _run_job(), revert it.

There's no job log in _request_job(), log traceback to the regular log.

Fix a typo in a comment.

Signed-off-by: Jonathon Hall <jonathon.hall@puri.sm>
